verification/spec/connection-handshake/ConnectionHandshakeModule.tla

```diff
@@ -0,0 +1,353 @@
+       Describes the current height of the chain.  
+
+     - connection -- a connection record.
+       Captures all the details of the connection on this chain.
+       For a full description of a connection record, see the
+       'Environment.Connections' set.
+
+     - client -- a client record.
+       Specifies the state of the client running on this chain.
+       For a full description of a client and the initialization values
+       for this record, see the InitClients set below.  
+     
+ ***************************************************************************)   
+    store,
+    (* A buffer (Sequence) holding any message(s) incoming to this module. *)
+    inBuf,
+    (* A buffer (Sequence) holding outbound message(s) from this module. *)
+    outBuf
+
+
+moduleVars == <<inBuf, outBuf, store>>
+
+
+(***************************************************************************
+    Helper operators.
+ ***************************************************************************)
+
+
+(* Returns true if 'para' matches the parameters in the local connection,
+    and returns false otherwise.
+ *)
+CheckLocalParameters(para) ==
+    LET local == store.connection.parameters.localEnd
+        remote == store.connection.parameters.remoteEnd 
+    IN /\ local.connectionID = para.localEnd.connectionID   
+       /\ remote.connectionID = para.remoteEnd.connectionID
+       /\ local.clientID = para.localEnd.clientID       
+       /\ remote.clientID = para.remoteEnd.clientID
+
+
+(* Validates a ConnectionParameter.
+    Expects as input a ConnectionParameter 'para' and returns true or false.
+    
+    This is a basic validation step, making sure that 'para' is valid with
+    respect to module-level constants ConnectionIDs and ClientIDs.
+    If there is a connection in the store, this also validates 'para'
+    against the parameters of an existing connection by relying on the
+    state predicate CheckLocalParameters.
+*)
+ValidConnectionParameters(para) ==
+    /\ para.localEnd.connectionID \in ConnectionIDs
+    /\ para.localEnd.clientID \in ClientIDs
+    /\ \/ store.connection = NullConnection
+       \/ /\ store.connection /= NullConnection
+          /\ CheckLocalParameters(para)
+
+
+(* Operator for reversing the connection ends.
+
+    Given a ConnectionParameters record 'para', returns a new set
+    of parameters where the local and remote ends are
+    flipped (i.e., reversed).
+ *)
+FlipConnectionParameters(para) ==
+    [localEnd   |-> para.remoteEnd,
+     remoteEnd  |-> para.localEnd]
+
+
+(* Operator for construcing a connection proof.
+
+    The connection proof is used to demonstrate to another chain that the
+    local store on this chain comprises a connection in a certain state. 
+ *)
+GetConnProof(connState) ==
+    [height |-> store.latestHeight,
+     connectionState |-> connState]
+
+
+(* Operator for construcing a client proof.
+ *)
+GetClientProof ==
+    [height |-> store.client.latestHeight]
```

Comment:
I think that you also need as with ConnProof to have additional field `height |-> store.latestHeight`. This effectively represents the proof and data in this case is clientState which we represent just with the `store.client.latestHeight`.

Reply:
It is also important to note that both proofs must have the same height. I would prefer to have it as a separate field in the message, that is you have 
```
{proof_height: <height_A>, 
 proofConn: <connection>, 
 proofConsensus: {
    consensus_height: <height_B>, 
    proofConsensus: <consensus>}
...
}
```

Reply:
Fixed after syncing with @ancazamfir.